### ISSUE TITLE
[BUG] make_reduce with unequal index set panels raises ValueError: Input X contains NaN 

### ISSUE THREAD

**Issue by EduardoPach** [sktime/sktime#4602]
**Describe the bug**

When using a panel set with unequal DatetimeIndex raises ValueError: Input X contains NaN. When changing so that the indexes are all the same no errors are raised.

**To Reproduce**

```python
import numpy as np
import pandas as pd
from sklearn.pipeline import make_pipeline
from sklearn.ensemble import RandomForestRegressor
from sktime.transformations.series.date import DateTimeFeatures
from sktime.transformations.series.summarize import WindowSummarizer
from sktime.forecasting.model_selection import temporal_train_test_split
from sktime.forecasting.compose import ForecastingPipeline, make_reduction

def create_df(cat: str, dates: pd.DatetimeIndex) -> pd.DataFrame:
    df = pd.DataFrame({
        'category': [cat]*len(dates),
        'date': dates,
        'value': np.random.randn(len(dates))
    })

    return df


dates_a = pd.date_range(start='2018-01-01', end="2019-01-01", freq='MS')
dates_b = pd.date_range(start='2018-04-01', end="2019-01-01", freq='MS')
dates_c = pd.date_range(start='2018-05-01', end="2019-01-01", freq='MS')

cats = ["a" ,"b", "c"]
dates = [dates_a, dates_b, dates_c]


df = pd.concat([create_df(cat, date) for cat, date in zip(cats, dates)], ignore_index=True)
df.set_index(["category", "date"], inplace=True)

train, test = temporal_train_test_split(df, test_size=0.2)

regressor = make_pipeline(RandomForestRegressor(random_state=1))

kwargs = {
    "lag_feature": {
        "lag": [1, 2, 3],
        "mean": [[1, 3]],
        "std": [[1, 3]]
    }
}

forecaster = make_reduction(
    regressor,
    transformers=[WindowSummarizer(**kwargs, n_jobs=1)],
    window_length=None,
    strategy="recursive",
    pooling="global",
    scitype="tabular-regressor"
)

pipe = ForecastingPipeline(
    steps=[
        ("daily_season", DateTimeFeatures(ts_freq="M")),
        ("forecaster", forecaster)
    ]
)

pipe.fit(train);
```

**Expected behavior**
That the pipeline would have been fitted without any error.

**Additional context**
The specific problem has only different starts for each panel, i.e. series start at different timepoints, but end at the same timepoint.

**Versions**
<details>

</details>

**Comment by EduardoPach** [comment]
Found that [_sliding_window_transform](https://github.com/sktime/sktime/blob/main/sktime/forecasting/compose/_reduce.py#L68) can't cope with unequal index when handling MultiIndex.

Explanation:

When we get the n_timepoints we are using the DatetimeIndex of the first panel that we got from [get_time_index](https://github.com/sktime/sktime/blob/main/sktime/datatypes/_utilities.py#L21), thus if the other panels aren't equal will use the wrong number of timepoints when doing [_cut_df(y, n_obs=n_timepoints-window_length)](https://github.com/sktime/sktime/blob/main/sktime/forecasting/compose/_reduce.py#L1518)

**Comment by fkiraly** [comment]
thanks for reporting - this should either be extended, and/or a clear user warning added as part of the `fit` logic

**Comment by kbpk** [comment]
I'd say the issue is even more significant. Just look at the `train` and `test` after split in the given example. There is the same [get_time_index](https://github.com/sktime/sktime/blob/77b48005bc13114c676f4743e87dd0878eb88b09/sktime/forecasting/model_selection/_split.py#LL1382C20-L1382C20) method used once.

**Comment by fkiraly** [comment]
@kbpk, yes, that is related to this issue: https://github.com/sktime/sktime/issues/4154

What is usually safer is to directly use the `split` method of a `cv splitter, or use `get_slice` and `get_window` from `datatypes._utilities`. This is a bit unclear for the user at the moment, and historically due to panel and hierarchical time series not supported initially, and then functionality for estimators being extended without the various loose methods keeping pace.

Help with this, e.g., patching `temporal_train_test_split` through to a splitter, or fixing it directly, would be appreciated.
